### PR TITLE
java.lang.Error should rollback the JTA transaction

### LINKED ISSUE #16455: JTA CDI should roll-back when intercepted method throws Error
Currently when a `@Transactional` method throws an `java.lang.Error` the transaction is not rolled-back.
But, the `java.lang.Error` is an un-checked exception and such behaviour is wrong and the transaction should be rolled-back.

This is an issue that was fixed in Narayana - WildFly integration by issue:
https://issues.redhat.com/browse/JBTM-3454
This issue should bring the same fix to Quarkus JTA CDI integration as well.

### On JTA spec and Errors

The JTA spec does not define the behaviour precisely for `@Transactional`. But from other spec or other parts of JTA spec we can consider the Error should rollback the transaction.

The JTA spec says about exceptions for `@Transactional` annotation:

```
By default checked exceptions do not result in the transactional interceptor marking
the transaction for rollback and instances of RuntimeException and its subclasses do. 
```

But from information about Synchronization at JTA spec

```
An unchecked exception thrown by a registered Synchronization object
causes the transaction to be aborted.
```

or about transaction handling withing EJB spec

```
all other exceptions [meant non-application-exceptions]
should "Mark the transaction for rollback."
```

### PR DESCRIPTION
**Pull request by ochaloup**
Fix #16455

### CODE REVIEW

File: extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/SneakyThrow.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package io.quarkus.narayana.jta.runtime;
+
+/**
+ * An utility class which makes possible to throw any exception as a {@link RuntimeException}.
+ * It means to throw checked exception (subtype of Throwable or Exception) as un-checked exception.
+ * This considers the Java 8 inference rule that states that a {@code throws E} is inferred as {@code RuntimeException}.
+ */
+public class SneakyThrow {
```
Comment (gastaldi) [inline extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/SneakyThrow.java:8]: Considering that this class is used only in this package, can you make it package-protected?

Comment (gastaldi) [reply extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/SneakyThrow.java:8]: Or even better, move the utility method to the `TransactionalInterceptorBase` class as a private method

Comment (ochaloup) [reply extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/SneakyThrow.java]: sure, done

**Review by manovotn — COMMENTED** [review]
@ochaloup would it be possible to also add a test?

**Comment by ochaloup** [comment]
@manovotn  here we go

**Review by manovotn — APPROVED** [review]
Thanks @ochaloup!

**Comment by famod** [comment]
Is this backport material?

**Comment by gastaldi** [comment]
@famod definitely backportable 👍🏻

**Pull request merged**